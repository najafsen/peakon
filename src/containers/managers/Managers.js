import React, { useEffect, useState } from "react";
import { useDispatch, useSelector } from "react-redux";
import { loadManagers } from "../../actions/managers.action";
import { selectFlattenedManagersItems } from "../../selectors/managers.selector";
import { Autocomplete } from "../../components/autocomplete/Autocomplete";
import './Managers.scss';

export const Managers = () => {
  const dispatch = useDispatch();
  const managers = useSelector(selectFlattenedManagersItems);
  const [manager, setManager] = useState(null);

  useEffect(() => {
    dispatch(loadManagers());
  }, [dispatch]);

  return (
    <div className="managers-component">
      <div className="managers-content">
        <label>Manager</label>
        <Autocomplete items={managers} onSelect={(manager) => setManager(manager)} />
        <h5>Selected manager in parent component: {manager ? manager.name : 'Not selected'}</h5>
      </div>
    </div>
  );
};
