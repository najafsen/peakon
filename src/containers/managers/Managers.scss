$base-size: 1rem;
$color: #576671;
$bg-color: #f7f6f4;

.managers-component {
  display: flex;
  justify-content: center;
  text-align: left;
  color: $color;

  .managers-content {
    flex: 0 1 400px;
    margin-top: 100px;
    background-color: $bg-color;
    padding: $base-size;

    label {
      display: inline-block;
      margin-bottom: 0.5 * $base-size;
    }
  }
}
