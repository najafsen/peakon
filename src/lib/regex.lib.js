export const escapeRegExp = (string) => {
    return string.replace(/[.*+?^${}()|[\]\\]/g, '\\$&');
}

export const createSearchRegex = (string, regexOptions = '') => {
    const regex = [...string].map(escapeRegExp).join('.*');
    return new RegExp(regex, regexOptions);
}