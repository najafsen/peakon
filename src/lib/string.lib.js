export const getNameInitials = (name) => {
    name = name || '';
    const nameParts = name.split(' ');
    
    if (nameParts.length >= 2) {
        return (nameParts[0][0] + nameParts[nameParts.length - 1][0]).toUpperCase();
    }

    return nameParts[0] ? nameParts[0][0].toUpperCase() : 'n/a';
}