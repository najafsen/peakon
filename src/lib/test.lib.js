import React from 'react';
import { FetchMock } from '@react-mock/fetch';
import { createStore, applyMiddleware } from 'redux';
import { Provider } from 'react-redux';
import thunk from 'redux-thunk';
import rootReducer from '../reducers';


export const withRedux = (ui, state = {}) => {
    const store = createStore(rootReducer, state, applyMiddleware(thunk));
    return (<Provider store={store}>{ui}</Provider>);
}

export const withFetchMock = (ui, fetchMockOptions = {}) => {
    return (<FetchMock {...fetchMockOptions}>{ui}</FetchMock>);
}