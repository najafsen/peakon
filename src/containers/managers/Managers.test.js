import React from 'react';
import { render } from '@testing-library/react';
import { Managers } from './Managers';
import { MANAGERS_INITIAL_STATE } from '../../reducers/managers.reducer';
import { withRedux, withFetchMock } from '../../lib/test.lib';
import employeesFixture from '../../fixtures/employees';

const url = 'https://gist.githubusercontent.com/daviferreira/41238222ac31fe36348544ee1d4a9a5e/raw/5dc996407f6c9a6630bfcec56eee22d4bc54b518/employees.json';

describe('Managers', () => {
    test('matches snapshot', () => {
        const state = {
            managers: MANAGERS_INITIAL_STATE
        };
        const fetchMock = {
            matcher: url,
            response: employeesFixture,
            method: 'GET'
        }
        const { asFragment } = render(
            withRedux(
                withFetchMock(<Managers />, { options: fetchMock }),
                state
            )
        );

        expect(asFragment()).toMatchSnapshot();
    });
});