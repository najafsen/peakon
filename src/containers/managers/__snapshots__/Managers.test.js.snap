// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Managers matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="managers-component"
  >
    <div
      class="managers-content"
    >
      <label>
        Manager
      </label>
      <div
        class="autocomplete-component"
      >
        <input
          data-testid="input"
          placeholder="Choose Manager"
          value=""
        />
        <i
          class="arrow-down"
          data-testid="arrow"
        />
      </div>
      <h5>
        Selected manager in parent component: Not selected
      </h5>
    </div>
  </div>
</DocumentFragment>
`;
